### PR TITLE
B sp 154 fix world scene creation

### PR DESCRIPTION
**Pull request by ethanfowler**
## Proposed changes

Please explain the changes you made. Add pictures or videos to explain them better if appropriate.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [x] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [x] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

**Review by BenStarmerSmith — COMMENTED** [review]
Updated year

**Review by georgiablanco — APPROVED** [review]

**Review by jeferal — APPROVED** [review]

File: sr_world_generator/src/sr_world_generator/clean_world_file.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
```
Comment (bdenoun) [inline sr_world_generator/src/sr_world_generator/clean_world_file.py:1]: output type hint missing in the whole file

File: sr_world_generator/README.md [hunk 2]
```diff
@@ -1,3 +1,3 @@
 # sr_world_generator
 
-This package contains code for creating, modifying and saving new `.world` and `.scene` files. For detailed instructions on how to use it, please see [this page](https://shadow-experimental.readthedocs.io/en/latest/user_guide/1_6_software_description.html#creating-a-new-world-scene).
+A lot of the code here is deprecated, since we changed our `.world` and `.scene` generation method. We now simply run a simulated robot as normal, rely on `gazebo2rviz` populating Rviz from Gazebo, save the world and scene files using Gazebo and Rviz's GUIs respectively, then clean the world file using [clean_world_file.launch](launch/clean_world_file.launch).
```
Comment (dg-shadow) [inline sr_world_generator/README.md]: Should this comment explain why we keep the code?

File: sr_world_generator/src/sr_world_generator/clean_world_file.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        Loads the world file from the specified path.
+
+        Args:
+            file_path: The path to the world file to load.
+        """
+        rospy.loginfo(f'Loading world file from {file_path}')
+        # Check if the file exists
+        if not os.path.isfile(file_path):
+            rospy.logerr(f'World file does not exist: {file_path}')
+            return False
```
Comment (dg-shadow) [inline sr_world_generator/src/sr_world_generator/clean_world_file.py:47]: I thought using multiple returns for flow control was frowned on?

File: sr_world_generator/src/sr_world_generator/clean_world_file.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    dry_run_param = rospy.get_param("~dry_run")
+    # Create a new WorldFileCleaner object
+    world_file_cleaner = WorldFileCleaner(dry_run_param)
+    # Load the world file, remove the models and state, and save the world file
+    if not world_file_cleaner.load_world_file(input_file_path_param):
+        sys.exit(0)
+    world_file_cleaner.remove_models(removed_model_names_param)
+    world_file_cleaner.remove_state()
+    if not world_file_cleaner.save_world_file(output_file_path_param):
+        sys.exit(0)
```
Comment (dg-shadow) [inline sr_world_generator/src/sr_world_generator/clean_world_file.py]: if we're expecting to have the output of this script used by something else, shouldn't it return a success if all goes well?

File: sr_world_generator/src/sr_world_generator/clean_world_file.py (hunk 1 above)
Comment (ethanfowler) [reply sr_world_generator/src/sr_world_generator/clean_world_file.py:1]: Addressed in https://github.com/shadow-robot/sr_tools/pull/63/commits/21a28bc4b564f6a686881c813fe4a8e53bf3d7af

File: sr_world_generator/README.md (hunk 2 above)
Comment (ethanfowler) [reply sr_world_generator/README.md]: Addressed in https://github.com/shadow-robot/sr_tools/pull/63/commits/21a28bc4b564f6a686881c813fe4a8e53bf3d7af

File: sr_world_generator/src/sr_world_generator/clean_world_file.py (hunk 3 above)
Comment (ethanfowler) [reply sr_world_generator/src/sr_world_generator/clean_world_file.py:47]: Not that I know of. Why would it be? Not sure what the alternative is.

File: sr_world_generator/src/sr_world_generator/clean_world_file.py (hunk 4 above)
Comment (ethanfowler) [reply sr_world_generator/src/sr_world_generator/clean_world_file.py]: Addressed in https://github.com/shadow-robot/sr_tools/pull/63/commits/21a28bc4b564f6a686881c813fe4a8e53bf3d7af

**Review by bdenoun — APPROVED** [review]

**Pull request merged**